### DISCUSSION TITLE
SignIn Error

### DISCUSSION THREAD

**Discussion by FranzPollany** [AdhocAdam/smletsexchangeconnector#419] [Q&A] [answered]
Hello,
I'm trying to set up the exchange connector. I used the Wiki and followed "Scenario 3: Exchange Online / Office 365 with OAuth".
I'm at the point where I see in azure / enterprise application the logon attempts with my SCSM workflow user.
But all are failed with error code 50126 (username or password wrong). I'm sure though, that the credentials are right.
Anyone any idea where I can look for an error?

**Comment by AdhocAdam** [comment]
Couple things. Can you confirm...
- In that section, the workflow account has the app you configured in its list of consented apps?
- That you don't have any warnings/errors in the Windows Event Log citing potential M365 issues

Assuming both of these things are okay, you can also manually attempt to verify authentication by using the [OAuth connection commands from the connector](https://github.com/AdhocAdam/smletsexchangeconnector/blob/9bcd5844bc92810b552a2c1967bd591813161e5c/smletsExchangeConnector.ps1#L4368-L4375) which I've ever so slightly re-arranged here for usability:

```powershell
#define
$AzureTenantID = ""
$AzureClientID = ""
$ewsUsername = ""
$ewsDomain = ""

#request an access token from Azure
$ReqTokenBody = @{
    Grant_Type    = "Password"
    client_Id     = $AzureClientID
    Username      = $ewsUsername + "@" + $ewsDomain
    Password      = $ewspassword
    Scope         = "https://outlook.office.com/EWS.AccessAsUser.All"
}

#get an access token
Invoke-RestMethod -Uri "https://login.microsoftonline.com/$AzureTenantID/oauth2/v2.0/token" -Method "POST" -Body $ReqTokenBody
```

- If you successfully get an access token, then there is a discrepancy between how you've entered the credentials in the above script vs. how you have them stored in SCSM 
- If you do _not_ get an access token, the error message should provide some context as to the nature of the fault

**Comment by FranzPollany** [comment, marked as answer]
Hello Adam, 
thank you for your reply.
I figured it out - the problem was/is the AD/ADFS infrastrukture. I use now an Azure AD Cloud User as Workflow User.
Now it works.
